### PR TITLE
🛠 Nx workspaces migration 🛠

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

1. Created a new project from scratch using https://nx.dev/packages/react/generators/application (https://github.com/cowprotocol/cowswap/pull/2965/commits/56da7f80e7b9b79caa912d8897e2e8dc3a7dfc6e)
2. Changed CRA+Craco to Vite and git-moved src files (https://github.com/cowprotocol/cowswap/pull/2965/commits/c80a73d6396b2d873e39941cc0b787c609b5602f)
3. Updated `@lingui/*`, `@@safe-global/*` and some other deps (https://github.com/cowprotocol/cowswap/pull/2965/commits/a816a94303dfd2c40f5b28ca523993bdf271db31)
4. Migrated eslint configs (https://github.com/cowprotocol/cowswap/pull/2965/commits/9a1db2d9ed0fbc26ff213d27a7ed04e15c008719)
5. Deleted paraswap API usage, because it was installed from `npm:@nenad91/paraswap#5.1.0` (https://github.com/cowprotocol/cowswap/pull/2965/commits/c80a73d6396b2d873e39941cc0b787c609b5602f#diff-6612c50e12e94bf6adb86bf89869de42844ad86ea62a0f2bd3e114b4a6e8a997R97)
6. Changed service-worked setup using https://vite-pwa-org.netlify.app/
7. Removed usage of `wearerequired/lint-action@v2` because it's not capable with NX workspaces
8. Updated cypress CI configs to make it work with NX
9. Changed `@lingui` setup corresponding to https://lingui.dev/ref/vite-plugin

### CODE REVIEW

File: apps/cowswap-frontend/vite.config.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      VitePWA({
+        registerType: 'autoUpdate',
+        injectRegister: 'auto',
+        devOptions: {
+          enabled: true,
+        },
+      }),
+    ],
+
+    // Uncomment this if you are using workers.
```
Comment (alfetopito) [inline apps/cowswap-frontend/vite.config.ts:76]: Aren't we?

File: package.json [hunk 2]
```diff
@@ -178,102 +156,96 @@
...
     "eslint-plugin-jsx-a11y": "6.7.1",
     "eslint-plugin-react": "7.32.2",
     "eslint-plugin-react-hooks": "4.6.0",
-    "eslint-plugin-unused-imports": "^2.0.0",
-    "https-browserify": "^1.0.0",
-    "husky": "^8.0.3",
-    "jest": "^29.5.0",
-    "jest-environment-jsdom": "^29.5.0",
+    "eslint-plugin-unused-imports": "^3.0.0",
+    "husky": "^8.0.0",
```
Comment (alfetopito) [inline package.json:227]: Nitpick: downgraded husky?

File: package.json [hunk 3]
```diff
@@ -178,102 +156,96 @@
...
+    "isomorphic-fetch": "^3.0.0",
     "jest-fetch-mock": "^3.0.3",
     "jest-styled-components": "^7.1.1",
-    "jsdom": "~20.0.3",
-    "launchdarkly-js-client-sdk": "^3.1.2",
```
Comment (alfetopito) [inline package.json:255]: Removing launchdarkly client?

File: package.json [hunk 4]
```diff
@@ -178,102 +156,96 @@
...
-    "typechain": "^8.1.1",
-    "verdaccio": "^5.0.4",
-    "vite": "^4.3.4",
-    "vite-plugin-dts": "~1.7.1",
-    "vite-plugin-eslint": "^1.8.1",
-    "vite-tsconfig-paths": "^4.0.2",
-    "vitest": "^0.31.0",
-    "webpack-bundle-analyzer": "^4.7.0"
+    "ts-node": "^10.9.1",
+    "typescript": "~5.1.3",
```
Comment (alfetopito) [inline package.json:241]: 🥳

**Review by alfetopito — APPROVED** [review]
Quick scan on top of the changes.
Have not tested it.
Looks ok, just some minor comments/questions

File: apps/cowswap-frontend/vite.config.ts (hunk 1 above)
Comment (shoom3301) [reply apps/cowswap-frontend/vite.config.ts:76]: Workers mean `Web workers` here.
For `Service workers` we use `import { VitePWA } from 'vite-plugin-pwa'`

File: package.json (hunk 2 above)
Comment (shoom3301) [reply package.json:227]: Fixed

File: package.json (hunk 3 above)
Comment (shoom3301) [reply package.json:255]: It still here:
https://github.com/cowprotocol/cowswap/blob/develop/package.json#L255
Actually we use `launchdarkly-react-client-sdk`

**Comment by anxolin** [comment]
Great job! 🔥

For now, i just tried to run this, and i noticed these logs. Is this expected?

image

**Comment by shoom3301** [comment]
>Also, were we getting this one before?

`yarn run prebuild` should help
@anxolin

**Comment by shoom3301** [comment]
@anxolin about `workbox` logs - there was an enabled flag for debugging, removed it

**Comment by elena-zh** [comment]
Testing is still in Progress, but I will post here my early notes and issues I face: 
1. Bothering me: PR link can be blocлed by an adblocker. Can we do anything to avoid this blocking? 
Screenshot_2
2. Connected user, no open orders/order history on the Limit/Advanced orders pages: 
image
3. Unsupported label is broken on a token list 
image

**Comment by shoom3301** [comment]
@elena-zh fixed all besides 1st point.
This issue is related to Service Worker cache, continue investigating it

**Comment by shoom3301** [comment]
@elena-zh The first one seems to be fixed as well, but you have to make a hard refresh at least once, after that it will not be mandatory

**Comment by shoom3301** [comment]
@elena-zh 
1. Fixed
2. This case is not stable and I can even reproduce it on prod, I don't think it was introduced in this PR
3. Fixed

**Review by elena-zh — APPROVED** [review]

**Pull request merged**